scripts/vmpublish.rb

```diff
@@ -0,0 +1,80 @@
+require 'logger'
+require 'pathname'
+require 'trollop'
+require 'json'
+
+require_relative 'config'
+
+
+def find_imgfac_image(image)
+  f = File.new(image)
+  Dir.chdir(STORAGE_DIR) do
+    Dir.glob('*.body') do |filename|
+      return filename.split('.').first if File.new(filename).stat.ino == f.stat.ino
+    end
+  end
+end
+
+
+def upload_imgfac_image(uuid, target, reference, version)
+  if target == 'vagrant'
+    creds_file = CREDS_DIR.join('atlas.json')
+    params = "atlas @ignored #{creds_file} --id #{uuid}"
+    params += " --parameter atlas_box_name #{reference} --parameter atlas_box_version #{version}"
+  elsif target == 'gce'
+    creds_file = CREDS_DIR.join('gce.json')
+    params = "gce @manageiq #{creds_file} --id #{uuid}"
+    params += " --parameter gce_object_name manageiq-#{reference}-#{version}.tar.gz"
+    params += " --parameter gce_image_name manageiq-#{reference}-#{version} --parameter gce_image_family centos-7"
```

Comment:
We may not gain a whole lot by using AwesomeSpawn here. Because of how the parameters are listed, we will likely just have something like `:params => %w(all the params here)` which doesn't do a lot for readability. Also we don't seem to be doing much escaping here (the other reason to use it) so I feel like we could keep it as is.